File: Python/min_max/min_max.py
def check_array_monotype(arr, passed_type):
    for item in arr:
        if type(item) is not passed_type: raise Exception(f'Cannot compare {passed_type} with {type(item)}.')

def min(*args):
    check_array_monotype(args, int)
    print(sorted(args)[0])

def max(*args):
    check_array_monotype(args, int)
    print(sorted(args)[len(args) - 1])

min(-3333333333333, 100_000_000, 10_000, 100_000, 1, -22222)
max(-3333333333333, 100_000_000, 10_000, 100_000, 1, -22222)

File: JavaScript/minMax/minMax.js
function checkArrayMonotype(arr, type) {
    arr.forEach( (arg, index) => {
        if (typeof arg !== type) throw new Error(`Cannot compare numbers with ${typeof arg} '${arg}' @ index ${index}.`)
    })
}

function min(...args) {
    checkArrayMonotype(args, "number");
    console.log(args.sort()[0]);
}

function max(...args) {
    checkArrayMonotype(args, "number");
    console.log(args.sort()[args.length - 1]);
}

max(100_000_000, '10', '2', 10_000, 100_000)
